src/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -1702,96 +1827,225 @@ public static IEnumerable<object[]> IndexOf_NonGeneric_TestData()
+            yield return new object[] { new NonGenericClass1[] { classObject, new NonGenericClass1() }, classObject, 0, 2, 0 };
+            yield return new object[] { new NonGenericClass1[] { classObject, new NonGenericClass1() }, new NonGenericClass1(), 0, 2, -1 };
+            yield return new object[] { new NonGenericClass1[] { classObject, new NonGenericClass1() }, classObject, 2, 0, -1 };
+            yield return new object[] { new NonGenericClass1[] { classObject, new NonGenericClass1() }, classObject, 0, 0, -1 };
+
+            // Struct
+            NonGenericStruct structObject = new NonGenericStruct();
+            yield return new object[] { new NonGenericStruct[] { structObject, new NonGenericStruct() }, structObject, 0, 2, 0 };
+            yield return new object[] { new NonGenericStruct[] { structObject, new NonGenericStruct() }, new NonGenericStruct(), 0, 2, 0 };
+            yield return new object[] { new NonGenericStruct[] { structObject, new NonGenericStruct() }, structObject, 2, 0, -1 };
+            yield return new object[] { new NonGenericStruct[] { structObject, new NonGenericStruct() }, structObject, 0, 0, -1 };
+
+            // Interface
+            ClassWithNonGenericInterface1 interfaceObject = new ClassWithNonGenericInterface1();
+            yield return new object[] { new NonGenericInterface1[] { interfaceObject, new ClassWithNonGenericInterface1() }, interfaceObject, 0, 2, 0 };
+            yield return new object[] { new NonGenericInterface1[] { interfaceObject, new ClassWithNonGenericInterface1() }, new ClassWithNonGenericInterface1(), 0, 2, -1 };
+            yield return new object[] { new NonGenericInterface1[] { interfaceObject, new ClassWithNonGenericInterface1() }, interfaceObject, 2, 0, -1 };
+            yield return new object[] { new NonGenericInterface1[] { interfaceObject, new ClassWithNonGenericInterface1() }, interfaceObject, 0, 0, -1 };
+
+            // Object
+            yield return new object[] { new object[] { new EqualsOverrider { Value = 1 } }, null, 0, 1, -1 };
+            yield return new object[] { new object[] { new EqualsOverrider { Value = 1 } }, new EqualsOverrider { Value = 1 }, 0, 1, 0 };
+            yield return new object[] { new object[] { new EqualsOverrider { Value = 1 } }, new EqualsOverrider { Value = 2 }, 0, 1, -1 };
+            yield return new object[] { new object[1], null, 0, 1, 0 };
+            yield return new object[] { new object[2], null, 2, 0, -1 };
+            yield return new object[] { new object[2], null, 0, 0, -1 };
+        }
+
+        public static IEnumerable<object[]> IndexOf_Dummy_SZArray_TestData()
+        {
+            yield return new object[] { new string[] { "Hello", "Hello", "Goodbye", "Goodbye" }, new object(), 0, 4, -1, "dummyString" };
+            yield return new object[] { new string[] { "Hello", "Hello", "Goodbye", "Goodbye" }, null, 0, 4, -1, "dummyString" };
+        }
```

Comment:
As you can see, I've tried avoiding having dummies as much as possible in these new tests.
The problem here is the following:
- xunit can't infer T from T[] (I've fixed this in a PR which hasn't been reviewed yet)
- xunit can't under T from null (defaults as object)

All of the other tests don't need dummies, except for null-as-T. Once my xunit PR xunit/xunit#965 is merged we will probably never need dummies again. Until then, this is the best we can do :(